include/onnxruntime/core/framework/data_types.h

```diff
@@ -59,50 +59,10 @@ struct MLFloat16 {
   explicit MLFloat16(uint16_t x) : val(x) {}
   explicit MLFloat16(float f);
 
-  // Taken from https://stackoverflow.com/a/60047308/12627730
-  float AsFloat(uint32_t x) const {
-    float out = 0.0f;
-    std::memcpy(&out, &x, sizeof(x));
-    return out;
-  }
-
-  // Taken from https://stackoverflow.com/a/60047308/12627730
-  uint32_t AsUint(float x) const {
-    uint32_t out = 0;
-    std::memcpy(&out, &x, sizeof(x));
-    return out;
-  }
-
-  float HalfToFloat(const uint16_t x) const {
```

Comment:
this implementation does not handle infinity and NaN correctly. using math::halfToFloat() instead.

Reply:
Was there a reason we didn't use math::halfToFloat() initially?

---
In reply to: [568288031](https://github.com/microsoft/onnxruntime/pull/6509#discussion_r568288031) [](ancestors = 568288031)

Reply:
i'm not sure. but math::halfToFloat() has quite a few uses and seems to be more correct than this implementation.

---
In reply to: [569858752](https://github.com/microsoft/onnxruntime/pull/6509#discussion_r569858752) [](ancestors = 569858752,568288031)